### ISSUE TITLE
Linux: context menu runs action under mouse when opened - New in 1.92.0

### ISSUE THREAD

**Issue by powellnorma** [microsoft/vscode#224743]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.92.0
- OS Version: Linux

So I am aware of https://github.com/microsoft/vscode/issues/113175 - However I only have this problem when using 1.92.0 - Downgrading to 1.91.1 fixes it. I noticed the context-menu now "fades in", whereas before it just appeared (with no animations). Were there some changes in 1.92.0 that could cause this?

Steps to Reproduce:

1. Use 1.92.0
2. In file browser, right-click somewhere - It opens the context-menu in such a way, that on release of the right-click it immediately executes the entry that's below the cursor. Optimally, the context-menu would appear with an offset such that this does not happen

**Comment by SiebelsTim** [comment]
I have the same behaviour. Furthermore, right-clicking shortly shows the menu and immediately closes the menu. Probably because releasing the right-click tries to select something. However, I cannot reproduce this on every click, few times it works. I think it depends on where you click. 

![Peek 2024-08-05 09-44](https://github.com/user-attachments/assets/0b48eb22-8635-4bcd-9cfe-f624a02d67dc)

**Comment by dancysoft** [comment]
Same.  I've been using VS Code on Linux without problems for years but as of 1.92.0 it exhibits the very dangerous/disruptive behavior of executing whatever command happens to be under the mouse when I right-click and release.

**Comment by spech66** [comment]
I have the same Problem with the latest update. Sometimes the menu stays open but most of the time the item directly on the mouse position is selected immediately. Downgrade to 1.91.1 fixes the problem.

**Comment by desabel** [comment]
Same. Kernel 6.1.0, Debian 12.1, KDE Plasma 5.27.5

**Comment by MerlijnElderhuis** [comment]
Same here. 1.91.1 has no problem.

**Comment by SiebelsTim** [comment]
@sbatten can we add the "new release" label again?

**Comment by mwilliamson** [comment]
I think this is related to https://github.com/microsoft/vscode/issues/224515, which provides a workaround of changing the setting "Window: Title Bar Style" to "custom" (although that does change the title bar).

**Comment by deefdragon** [comment]
Its related, but I think unique. I encounterd this issue previously, but fixed it (dont remember how, IE if I followed the fix mentioned in the other thread) several years ago. I recently upgraded my vscode, and have started encountering it once again.

**Comment by fxzxmic** [comment]
I believe this is simply due to the fact that both mouse presses and releases are used as conditions to trigger the action.

I'm using Linux and this happens when `window.titleBarStyle` is set to custom or native.

**Comment by erentar** [comment]
@fxzxmic yes, having the custom titlebar also themes the right click menu, and this issue no longer happens.

the problem presents when the native titlebar is selected, rather than the custom

**Comment by fweth** [comment]
Could we maybe get the option to select only a native title bar without a native context menu?

**Comment by lah7** [comment]
@fweth There is a pull request open for splitting the options. It was initiated to fix #224515.

* https://github.com/microsoft/vscode/pull/229385
* https://github.com/microsoft/vscode/issues/229053

I think calling them "native context menus" is inaccurate, since they're Electron/Chromium based menus.<sup>[[1]](https://github.com/electron/electron/pull/45033)</sup> Only Electron under macOS use native menus, although #213448 shows this bug happening on macOS too. Menu regressions started since [VS Code 1.92.0 bumped to Electron 30](https://code.visualstudio.com/updates/v1_92#_electron-30-update), so it _could_ be an upstream bug.

What display server is everyone using here? X11 or Wayland? Just in case it's specific to one of them.  I'm currently using 1.98.2 with custom menus<sup>[[2]](https://github.com/microsoft/vscode/issues/224515#issuecomment-2545724749)</sup> under X11 which isn't experiencing this issue.

**Comment by fweth** [comment]
I'm using X11 (XFCE). I don't know where the context menu comes from, it looks different from the XFCE context menus, more like GNOME.

**Comment by fweth** [comment]
If I'm not mistaken, this has been fixed in `1.100.0-insider` 🥳

**Comment by deepak1556** [comment]
This is tracked https://github.com/electron/electron/issues/46163

**Comment by Makskuk** [comment]
Faced with same issue after update vscode. "Window: Title Bar Style" is "native"

Версия: 1.99.1
Фиксация: 7c6fdfb0b8f2f675eb0b47f3d95eeca78962565b
Дата: 2025-04-04T15:58:59.624Z
Electron: 34.3.2
ElectronBuildId: 11161073
Chromium: 132.0.6834.210
Node.js: 20.18.3
V8: 13.2.152.41-electron.0
ОС: Linux x64 6.8.0-57-generic

name -a
Linux kuklin-mint 6.8.0-57-generic #59~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Mar 19 17:07:41 UTC 2 x86_64 x86_64 x86_64 GNU/Linux


Menu not disappearing if cursor not in any of menu items at moment of menu drawing.
